### PR TITLE
Disallow modifiers in object literal property assignment

### PR DESCRIPTION
**Pull request by RyanCavanaugh**
Fixes bug #5994

### CODE REVIEW

**Comment by mhegazy** [comment]
looks good.

**Comment by RyanCavanaugh** [comment]
Need to revise since `async` is valid

**Comment by DanielRosenwasser** [comment]
@RyanCavanaugh I dont see tests for `async` though.

**Comment by RyanCavanaugh** [comment]
@DanielRosenwasser https://github.com/Microsoft/TypeScript/blob/master/tests/cases/conformance/async/es6/asyncAwait_es6.ts#L24

**Comment by sandersn** [comment]
:+1:

File: src/compiler/checker.ts [hunk 1]
```diff
@@ -15969,6 +15969,16 @@ namespace ts {
                     return grammarErrorOnNode((<ShorthandPropertyAssignment>prop).equalsToken, Diagnostics.can_only_be_used_in_an_object_literal_property_inside_a_destructuring_assignment);
                 }
 
+                // Modifiers cannot appear in property assignments
+                forEach(prop.modifiers, mod => {
+                    if (mod.kind !== SyntaxKind.AsyncKeyword) {
+                        grammarErrorOnNode(mod, Diagnostics._0_modifier_cannot_be_used_here, getTextOfNode(mod));
```
Comment (vladima) [inline src/compiler/checker.ts]: looks like body of `then` statements is the same in both `if` statements, can you merge them?

**Pull request merged**